### PR TITLE
fix: Multichain: UX: Check for transactions on all networks and QueuedRequestCount

### PR DESCRIPTION
**Pull request by darkwing**
## **Description**

There's a case where there could be a race condition between the `QueuedRequestController`'s network switching and the Routes' network switching.  The root issue is that the use of `getUnapprovedTransactions` only gets transactions on the `current` chain, thus triggering Routes to switch networks once the last transaction of the current chain is done, but there could be more transactions on other chains which we should allow QRC to handle network switching for.

Thus, this PR lets QRC control network switching until there are absolutely no transactions or queued requests left.

[This PR includes an **E2E](https://github.com/MetaMask/metamask-extension/pull/25536)** for the scenario where there are multiple queued transactions but since this is a race issue, it's difficult to reproduce in a provable E2E

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25614?quickstart=1)

## **Related issues**

Possibly related: https://github.com/MetaMask/metamask-extension/issues/25596

## **Manual testing steps**

1. Open `Tab 1`, connect to `Uniswap` on `Ethereum Mainnet`
2. Open `Tab 2`, connect to `PancakeSwap` on `BNB Chain`
3. Open `Tab 3`, connect to `Test Dapp` on `Sepolia`
4. Initiate a swap on `Tab 1` and `Tab 2` *BUT DO NOT CONFIRM IT, JUST MOVE ON TO THE NEXT TAB*
5. Initiate a send on `Tab 3` *BUT DO NOT CONFIRM IT*
6. On the confirmation screen, you should still see the first confirmation from `Tab 1` (`Uniswap`) on `Ethereum Mainnet`; confirm or reject it.  See the confirmation window close
7. A new confirmation popup should come up with the `PancakeSwap`/ `Tab 2` confirmation on `BNB` chain; confirm or reject it.  See the confirmation window close
8. See one last confirmation screen pop up for the `Tab 3` / `Test Dapp` send on `Sepolia`.  Confirm or reject it.

Create transactions on different networks, reject and/or confirm them, click around between popup and notification window, ensure nothing unexpected occurs

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/selectors/selectors.js [hunk 1]
```diff
@@ -1778,7 +1779,9 @@ export function getShowRecoveryPhraseReminder(state) {
  * @returns Number of unapproved transactions
  */
 export function getNumberOfAllUnapprovedTransactionsAndMessages(state) {
```
Comment (adonesky1) [inline ui/selectors/selectors.js:1781]: can we also rename [this variable](https://github.com/MetaMask/metamask-extension/blob/03b5d5ef766f05f20ddf5efa90b129f7944d5af4/ui/pages/routes/routes.container.js#L137) to `numberOfAllUnapprovedTransactionsAndMessages`?

**Review by adonesky1 — DISMISSED** [review]
LGTM

File: ui/selectors/transactions.js [hunk 2]
```diff
@@ -56,15 +59,22 @@ export const getCurrentNetworkTransactions = createDeepEqualSelector(
...
-        return result;
-      }, {});
+// Unlike `getUnapprovedTransactions` and `getCurrentNetworkTransactions`
+// returns the total number of unapproved transactions on all networks
+export const getAllUnapprovedTransactions = createDeepEqualSelector(
+  (state) => {
+    const { transactions } = state.metamask || [];
+    if (!transactions?.length) {
+      return [];
+    }
```
Comment (jiexi) [inline ui/selectors/transactions.js:74]:
```suggestion
    const { transactions } = state.metamask ?? [];
    if (!transactions.length) {
      return {};
    }
```

**Comment by jiexi** [comment]
LGTM after the one comment

File: ui/selectors/transactions.js (hunk 2 above)
Comment (jiexi) [reply ui/selectors/transactions.js:74]: feel free to ignore the nullish coalesce and optional chaining, but returning the empty object is necessary

**Review by jiexi — DISMISSED** [review]

**Review by adonesky1 — DISMISSED** [review]

**Review by salimtb — DISMISSED** [review]

File: ui/selectors/selectors.js (hunk 1 above)
Comment (Gudahtt) [inline ui/selectors/selectors.js:1781]: I see that this selector is used for the `routes` prop `unapprovedTransactions`. This prop name was already confusing before now for a few reasons: it's just the count, not the transactions themselves, and it's not just transactions (it includes non-transaction confirmations).

But it's slightly more confusing now that we're including unapproved confirmations across all chains. Perhaps we should take this opportunity to make the prop name more clear? e.g. something like `totalUnapprovedConfirmationCount`

Comment (darkwing) [reply ui/selectors/selectors.js:1781]: Updated!

**Review by jiexi — APPROVED** [review]

**Review by adonesky1 — APPROVED** [review]

**Pull request merged**